### PR TITLE
Make generator work with latest version of Kapitan

### PR DESCRIPTION
**Pull request by ademariag**
Make the generator work with the latest version of Kapitan/Kadet.

Requires https://github.com/kapicorp/kapitan/pull/845 (kapitan >= 0.31)

Based on https://github.com/kapicorp/kapitan-reference/pull/127

### CODE REVIEW

File: components/generators/argocd/__init__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def j2(filename, ctx):
     return render_jinja2_file(filename, ctx, search_paths=search_paths)
 
 
-inv = inventory()
+inv = inventory(lazy=True)
```
Comment (ramaro) [inline components/generators/argocd/__init__.py:18]: ideally, we'd want to not use lazy

File: components/generators/kubernetes/__init__.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 search_paths = args.get("search_paths")
 
-inv = inventory()
+inv = inventory(lazy=True)
```
Comment (ramaro) [inline components/generators/kubernetes/__init__.py:13]: ideally, we'd want to not use lazy

File: components/generators/argocd/__init__.py (hunk 1 above)
Comment (ademariag) [reply components/generators/argocd/__init__.py:18]: I intend to refactor it heavily to move to the new BaseClass, as part of that we might achieve this as well

File: components/generators/kubernetes/__init__.py (hunk 2 above)
Comment (ademariag) [reply components/generators/kubernetes/__init__.py:13]: I intend to refactor it heavily to move to the new BaseClass, as part of that we might achieve this as well

File: components/generators/rabbitmq/__init__.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ def j2(filename, ctx):
     return render_jinja2_file(filename, ctx, search_paths=search_paths)
 
 
-inv = inventory()
+inv = inventory(lazy=True)
```
Comment (ramaro) [inline components/generators/rabbitmq/__init__.py:18]: ideally, we'd want to not use lazy

File: components/generators/argocd/__init__.py (hunk 1 above)
Comment (ramaro) [reply components/generators/argocd/__init__.py:18]: got it

**Review by ramaro — APPROVED** [review]

**Pull request merged**